processing/src/main/java/io/druid/segment/incremental/IncrementalIndexAdapter.java

```diff
@@ -49,13 +50,15 @@
   private final Interval dataInterval;
   private final IncrementalIndex<?> index;
   private final Map<String, Map<String, MutableBitmap>> invertedIndexes;
+  private final Map<String, Object> metadata;
 
   public IncrementalIndexAdapter(
-      Interval dataInterval, IncrementalIndex<?> index, BitmapFactory bitmapFactory
+      Interval dataInterval, IncrementalIndex<?> index, BitmapFactory bitmapFactory, Map<String, Object> metadata
```

Comment:
Is there a reason to have metadata be part of this? It doesn't seem to be used for anything so it seems simpler to just have getMetaData always return null.